pkg/handler/commit.go

```diff
@@ -101,3 +102,87 @@ func (h *commitHandler) GetCommitByMessage(w http.ResponseWriter, r *http.Reques
 
 	_ = json.NewEncoder(w).Encode(resp)
 }
+
+func (h *commitHandler) CommitReleased(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	// Get request data from query params
+	request, errMessage := GetCommitReleasedRequest(r)
+	if errMessage != "" {
+		http.Error(w, errMessage, http.StatusBadRequest)
+		return
+	}
+
+	branchExists, err := checkIfBranchExists(request.Owner, request.Repository, request.ReleaseBranch)
+	if err != nil {
+		http.Error(w, "Error checking if branch exists: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	if !branchExists {
+		http.Error(w, "Branch does not exist in the repository", http.StatusBadRequest)
+		return
+	}
+
+	baseUrl := fmt.Sprintf("https://api.github.com/repos/%s/%s/commits?sha=%s&per_page=100&page=", request.Owner, request.Repository, request.ReleaseBranch)
+	method := "GET"
+
+	req, err := http.NewRequest(method, baseUrl, nil)
+	if err != nil {
+		http.Error(w, "Error generating new request: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	commitReleased := false
+
+	for page_number := 1; ; page_number++ {
+		url := baseUrl + strconv.Itoa(page_number)
+		u, err := urlpkg.Parse(url)
+		if err != nil {
+			http.Error(w, "Error generating new URL: "+err.Error(), http.StatusInternalServerError)
+			return
+		}
+		req.URL = u
+
+		// Define a variable of type []Commit to store the data
+		var commits []model.CommitData
+
+		client := &http.Client{}
+		res, err := client.Do(req)
+		if err != nil {
+			http.Error(w, "Error making request to GitHub: "+err.Error(), http.StatusInternalServerError)
+			return
+		}
+		defer res.Body.Close()
+
+		body, err := io.ReadAll(res.Body)
+		if err != nil {
+			http.Error(w, "Error reading response: "+err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		// Unmarshal JSON data into commits variable
+		err = json.Unmarshal(body, &commits)
+		if err != nil {
+			http.Error(w, "Error unmarshalling JSON: "+err.Error(), http.StatusInternalServerError)
+			return
+		}
```

Comment:
We are checking for errors and exiting the function in multiple places. It could be beneficial to refactor it into a separate function to make the code DRY?

Reply:
Agreed, thanks for the review. I have refactored the code to improve the readability